Qt: implement mobile UI shutdown

BlockMechanic opened the pull request:
Resolves shutdown issue for mobile UI.
~~connect(engine, &QQmlEngine::quit, this, &QApplication::quit)~~
becomes
connect(engine, &QQmlEngine::quit, qApp, &QApplication::quit)

Also prevents segfault caused by referencing uninitialzed pollShutdownTimer

BlockMechanic committed:
Fix Shutdown

icota commented on src/qt/bitcoin.cpp:
```diff
@@ -324,8 +326,9 @@ void BitcoinApplication::requestShutdown()
     // to complete an operation, like wait for a RPC execution to complete.
     window->setClientModel(nullptr);
+#ifndef MOBILE_GUI
     pollShutdownTimer->stop();
```
If you want to properly implement shutdown it's probably no good to ignore its timer... Take a look at what it does for the 'old' GUI and see if we need the same functionality (there is a `detectShutdown`). If it's really not needed then it seems to me that we can get away with just the one #ifdef here

BlockMechanic replied:
OK, will look into it.

icota replied:
Looks good! Can you please squash the commits into something called e.g. "Qt: implement mobile UI shutdown"?

BlockMechanic replied:
Ok, I squashed and it's now under suggested name.

BlockMechanic committed:
re-enable pollShutdownTimer

BlockMechanic committed:
Qt: implement mobile UI shutdown
modified:   src/qt/bitcoin.cpp
modified:   src/qt/bitcoinmobilegui.cpp
modified:   src/qt/bitcoinmobilegui.h
modified:   src/qt/res/qml/WalletPane.qml

BlockMechanic committed:
Merge branch 'fix-close-shutdown' of https://github.com/BlockMechanic/bitcoin into fix-close-shutdown
